smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/util/StringStore.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package software.amazon.smithy.typescript.codegen.util;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.LinkedList;
+import java.util.Map;
+import java.util.Queue;
+import java.util.Set;
+import java.util.TreeMap;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * Intended for use at the
+ * {@link software.amazon.smithy.typescript.codegen.integration.ProtocolGenerator.GenerationContext}
+ * level, this class allocates and tracks variables assigned to string literals, allowing a
+ * form of compression on long protocol serde files.
+ */
+public class StringStore {
+    private final ConcurrentHashMap<String, String> literalToVariable = new ConcurrentHashMap<>();
```

Comment:
since this class isn't actually thread safe at the moment, you could just use a TreeMap here directly, and a HashMap below.